### DISCUSSION TITLE
Refactor `withdraw` and `withdrawMax`

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/flow#54] [Ideas]
I would like suggest the following changes:

1. 
```solidity
// Rename
function withdrawMax(uint256 streamId, address to) external;
// to
function withdraw(uint256 streamId, address to) external;
```
**Rationale**: >90% of users would just like to withdraw maximum amount

2.
```solidity
// Rename
function withdraw(uint256 streamId, address to, uint40 time) external;
// to
function withdrawAt(uint256 streamId, address to, uint40 time) external;
```
**Rationale**: This should be treated as a special function that withdraws at a time t.

**Comment by andreivladbrg** [comment]
Interesting proposal

> 90% of users would just like to withdraw maximum amount

I think this might be right but we need first to make a query on dune to be sure

I am not in favor of renaming `withdrawMax` to `withdraw`

But we could rename `withdraw` to `withdrawAt`.

So:
```solidity 
function withdrawMax(uint256 streamId, address to) external;
function withdrawAt(uint256 streamId, address to, uint40 time) external;
```

Does it work for you?

**Reply by smol-ninja** [reply]
`withdrawAt` or `withdrawTo`? (I am in favour of `withdrawAt` and not in `withdrawTo`)

Not having `withdraw` works.

Similarly, in https://github.com/sablier-labs/v2-open-ended/discussions/53, `withdrawMaxMultiple` and `withdrawMultipleAt`?

**Reply by andreivladbrg** [reply]
`withdrawAt`, edited now, sorry

**Comment by smol-ninja** [comment]
As per the discussion between me and @andreivladbrg, we have decided to rename `withdraw` to `withdrawAt` and keep `withdrawMax` unchanged.

- https://github.com/sablier-labs/v2-open-ended/issues/72
